app/src/main/java/org/astraea/performance/Manager.java

```
@@ -61,7 +63,10 @@ public Manager(
     if (exeTime.percentage(payloadNum.getAndIncrement(), System.currentTimeMillis() - start)
         >= 100D) return Optional.empty();
 
-    byte[] payload = (this.fixedSize) ? new byte[size] : new byte[rand.nextInt(size) + 1];
+    byte[] payload =
+        (this.fixedSize)
+            ? new byte[dataSize.measurement(DataUnit.Byte).intValue()]
+            : new byte[rand.nextInt(dataSize.measurement(DataUnit.Byte).intValue()) + 1];
```

這邊的程式碼似乎假設假設 measurement 出來的數值不會超過 int 大小，但 BigInteger 可以很輕易的超出這個範圍，而且其 `intValue()` 的實作明顯指出超出範圍時回傳結果會是失去訊息，能否在前面加上一些檢查是否超出合理數字範圍，來做到 fail fast 的效果